diff -Naurb ./wine-1.7.52/dlls/winex11.drv/event.c ./wine-1.7.52-patched/dlls/winex11.drv/event.c
--- ./wine-1.7.52/dlls/winex11.drv/event.c	2015-10-02 17:20:05.000000000 +0300
+++ ./wine-1.7.52-patched/dlls/winex11.drv/event.c	2015-10-13 14:25:12.000000000 +0300
@@ -1036,7 +1036,7 @@
     if (!data->mapped || data->iconic) goto done;
     if (data->whole_window && !data->managed) goto done;
     /* ignore synthetic events on foreign windows */
-    if (event->send_event && !data->whole_window) goto done;
+    // if (event->send_event && !data->whole_window) goto done;
     if (data->configure_serial && (long)(data->configure_serial - event->serial) > 0)
     {
         TRACE( "win %p/%lx event %d,%d,%dx%d ignoring old serial %lu/%lu\n",
diff -Naurb ./wine-1.7.52/dlls/winex11.drv/window.c ./wine-1.7.52-patched/dlls/winex11.drv/window.c
--- ./wine-1.7.52/dlls/winex11.drv/window.c	2015-10-02 17:20:05.000000000 +0300
+++ ./wine-1.7.52-patched/dlls/winex11.drv/window.c	2015-10-13 15:59:29.968686454 +0300
@@ -1131,7 +1131,11 @@
             if (data->surface && data->vis.visualid != default_visual.visualid)
                 data->surface->funcs->flush( data->surface );
         }
-        else set_xembed_flags( data, XEMBED_MAPPED );
+        else {
+            XMapWindow( data->display, data->whole_window );
+            XFlush( data->display );
+            set_xembed_flags( data, XEMBED_MAPPED );
+        }
 
         data->mapped = TRUE;
         data->iconic = (new_style & WS_MINIMIZE) != 0;
